qt/scientific_interfaces/ISISReflectometry/GUI/Experiment/ExperimentPresenter.cpp

```diff
@@ -111,7 +111,7 @@ void ExperimentPresenter::notifyPreviewApplyRequested(PreviewRow const &previewR
   if (auto const foundRow = m_model.findLookupRow(previewRow, m_thetaTolerance)) {
     auto lookupRowCopy = *foundRow;
 
-    lookupRowCopy.setRoiDetectorIDs(previewRow.getSelectedBanks());
+    lookupRowCopy.setRoiDetectorIDs(getRangesFromListOfBanks(previewRow.getSelectedBanks()));
```

Comment:
I noticed that the selected banks are set in `PreviewPresenter::notifyInstViewShapeChanged()`, which formats them as a comma delimited string. Would it work to set the format we need straight away within the `PreviewPresenter` method instead of having to reformat here, or does that cause problems?